src/Activity-Repository.js

```diff
@@ -11,24 +11,12 @@ class Activity {
     return this.activityData.filter((activityObj) => activityObj.date === date);
   }
 
-  returnAvgStairsClimbedAllUsersByDate(date)  {
-    let allUsersStairs = this.findUserCurDate(date);
-    let allUsersStairsTotal = allUsersStairs.reduce((activityObjA, activityObjB) => activityObjA + activityObjB.flightsOfStairs, 0);
-    return parseInt(allUsersStairsTotal / allUsersStairs.length);
+  returnUserAvgsByDate(date, key)  {
+    let allUsersActivity = this.findUserCurDate(date);
+    let allUsersTotal = allUsersActivity.reduce((activityObjA, activityObjB) => activityObjA + activityObjB[key], 0);
+    return parseInt(allUsersTotal / allUsersActivity.length);
   }
 
-  returnAvgStepsTakenAllUsersByDate(date) {
```

Comment:
wow you were able to cut out alot. On my Friends: Weekly Step Challenge all the user data is the same. can you check that

Reply:
Thanks for the heads up here, I did some copy/pasting and apparently my eyes weren't totally open on that one. Fixed now, thanks for the extra set of eyes!